src/screens/History/HistoryListArchanova.js

```diff
@@ -76,7 +78,17 @@ export default HistoryListArchanova;
 // Extracted from legacy `Home` screen
 function useHistoryFeedItems(): any[] {
   const accounts = useRootSelector((root) => root.accounts.data);
-  const userEvents = useRootSelector((root) => root.userEvents.data);
+  const walletEvents = useRootSelector((root) => root.walletEvents.data);
+  const activeAccount = useActiveAccount();
+
+  const archanovaEthereumWalletEvents = activeAccount
+    ? walletEvents?.[getAccountId(activeAccount)]?.ethereum ?? []
+    : [];
+
+  const mappedArchanovaEthereumWalletEvents = archanovaEthereumWalletEvents.map((walletEvent) => ({
+    ...walletEvent,
+    createdAt: new Date(+parseDate(walletEvent.date) / 1000),
+  }));
```

Comment:
Consider using `useMemo` or extracting as memoized selector to avoid computation on every render.